lib/phlex/compiler/file_compiler.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Phlex::Compiler::FileCompiler < Prism::Visitor
+	def initialize(compiler)
+		@compiler = compiler
+	end
+
+	def compile(node)
+		visit(node)
+	end
+
+	def visit_class_node(node)
+		if @compiler.line == node.location.start_line
```

Comment:
It’s possible to define more than one class on a single line, so we should also check the name.